--- ./contrib/halog/Makefile.orig	2020-10-17 22:35:22.480132129 +0000
+++ ./contrib/halog/Makefile	2020-10-17 22:36:13.268471122 +0000
@@ -16,7 +16,7 @@
 OBJS     = halog
 
 halog: halog.c fgets2.c
-	$(CC) $(OPTIMIZE) $(DEFINE) -o $@ $(INCLUDE) $(SRC)/ebtree.c $(SRC)/eb32tree.c $(SRC)/eb64tree.c $(SRC)/ebmbtree.c $(SRC)/ebsttree.c $(SRC)/ebistree.c $(SRC)/ebimtree.c $^
+	$(CC) $(OPTIMIZE) $(CFLAGS) $(LDFLAGS) $(DEFINE) -o $@ $(INCLUDE) $(SRC)/ebtree.c $(SRC)/eb32tree.c $(SRC)/eb64tree.c $(SRC)/ebmbtree.c $(SRC)/ebsttree.c $(SRC)/ebistree.c $(SRC)/ebimtree.c $^
 
 clean:
 	rm -f $(OBJS) *.[oas]
--- ./contrib/ip6range/Makefile.orig	2020-10-17 23:01:12.760529299 +0000
+++ ./contrib/ip6range/Makefile	2020-10-17 23:01:59.993951109 +0000
@@ -7,7 +7,7 @@
 all: $(OBJS)
 
 %: %.c
-	$(CC) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.o *.a *~
--- ./contrib/iprange/Makefile.orig	2020-10-17 23:01:08.012865757 +0000
+++ ./contrib/iprange/Makefile	2020-10-17 23:01:52.194456116 +0000
@@ -7,7 +7,7 @@
 all: $(OBJS)
 
 %: %.c
-	$(CC) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OPTIMIZE) -o $@ $^
 
 clean:
 	rm -f $(OBJS) *.o *.a *~
