--- tree-1.8.0/tree.c   2018-01-10 18:47:14.000000000 +0200
+++ tree-1.8.0/tree.c  2021-09-22 13:53:15.496624444 +0300
@@ -114,7 +114,7 @@
   }

 /* Until I get rid of this hack, make it linux/cygwin/HP nonstop only: */
-#if defined (LINUX) || defined (CYGWIN) || defined (__TANDEM)
+#if defined (LINUX) || defined (CYGWIN) || defined (__TANDEM) || defined (__ANDROID__)
   mb_cur_max = (int)MB_CUR_MAX;
 #else
   mb_cur_max = 1;
