diff -Naur xorg-server-1.20.11/hw/xwayland/ashmem.h xorg-server-1.20.11/hw/xwayland/ashmem.h
--- xorg-server-1.20.11/hw/xwayland/ashmem.h	1970-01-01 03:00:00.000000000 +0300
+++ xorg-server-1.20.11/hw/xwayland/ashmem.h	2021-08-14 10:48:34.153323991 +0300
@@ -0,0 +1,47 @@
+/****************************************************************************
+ ****************************************************************************
+ ***
+ ***   This header was automatically generated from a Linux kernel header
+ ***   of the same name, to make information necessary for userspace to
+ ***   call into the kernel available to libc.  It contains only constants,
+ ***   structures, and macros generated from the original header, and thus,
+ ***   contains no copyrightable information.
+ ***
+ ****************************************************************************
+ ****************************************************************************/
+#ifndef _LINUX_ASHMEM_H
+#define _LINUX_ASHMEM_H
+
+#include <linux/limits.h>
+#include <linux/ioctl.h>
+#include <stdint.h>
+
+#define ASHMEM_NAME_LEN 256
+
+#define ASHMEM_NAME_DEF "dev/ashmem"
+
+#define ASHMEM_NOT_PURGED 0
+#define ASHMEM_WAS_PURGED 1
+
+#define ASHMEM_IS_UNPINNED 0
+#define ASHMEM_IS_PINNED 1
+
+struct ashmem_pin {
+ uint32_t offset;
+ uint32_t len;
+};
+
+#define __ASHMEMIOC 0x77
+
+#define ASHMEM_SET_NAME _IOW(__ASHMEMIOC, 1, char[ASHMEM_NAME_LEN])
+#define ASHMEM_GET_NAME _IOR(__ASHMEMIOC, 2, char[ASHMEM_NAME_LEN])
+#define ASHMEM_SET_SIZE _IOW(__ASHMEMIOC, 3, size_t)
+#define ASHMEM_GET_SIZE _IO(__ASHMEMIOC, 4)
+#define ASHMEM_SET_PROT_MASK _IOW(__ASHMEMIOC, 5, unsigned long)
+#define ASHMEM_GET_PROT_MASK _IO(__ASHMEMIOC, 6)
+#define ASHMEM_PIN _IOW(__ASHMEMIOC, 7, struct ashmem_pin)
+#define ASHMEM_UNPIN _IOW(__ASHMEMIOC, 8, struct ashmem_pin)
+#define ASHMEM_GET_PIN_STATUS _IO(__ASHMEMIOC, 9)
+#define ASHMEM_PURGE_ALL_CACHES _IO(__ASHMEMIOC, 10)
+
+#endif
diff -Naur xorg-server-1.20.11/hw/xwayland/xwayland-cursor.c xorg-server-1.20.11/hw/xwayland/xwayland-cursor.c
--- xorg-server-1.20.11/hw/xwayland/xwayland-cursor.c	2021-04-13 17:11:40.000000000 +0300
+++ xorg-server-1.20.11/hw/xwayland/xwayland-cursor.c	2021-08-14 10:48:34.153323991 +0300
@@ -66,7 +66,8 @@
     PixmapPtr pixmap;
 
     pixmap = xwl_shm_create_pixmap(screen, cursor->bits->width,
-                                   cursor->bits->height, 32, 0);
+                                   cursor->bits->height, 32,
+                                   CREATE_PIXMAP_USAGE_BACKING_PIXMAP);
     dixSetPrivate(&cursor->devPrivates, &xwl_cursor_private_key, pixmap);
 
     return TRUE;
diff -Naur xorg-server-1.20.11/hw/xwayland/xwayland-glamor-gbm.c xorg-server-1.20.11/hw/xwayland/xwayland-glamor-gbm.c
--- xorg-server-1.20.11/hw/xwayland/xwayland-glamor-gbm.c	2021-04-13 17:11:40.000000000 +0300
+++ xorg-server-1.20.11/hw/xwayland/xwayland-glamor-gbm.c	2021-08-14 10:48:34.154323986 +0300
@@ -202,9 +202,9 @@
     PixmapPtr pixmap = NULL;
 
     if (width > 0 && height > 0 && depth >= 15 &&
-        (hint == 0 ||
-         hint == CREATE_PIXMAP_USAGE_BACKING_PIXMAP ||
-         hint == CREATE_PIXMAP_USAGE_SHARED)) {
+        (hint == CREATE_PIXMAP_USAGE_BACKING_PIXMAP ||
+         hint == CREATE_PIXMAP_USAGE_SHARED ||
+         (xwl_screen->rootless && hint == 0))) {
         uint32_t format = gbm_format_for_depth(depth);
 
 #ifdef GBM_BO_WITH_MODIFIERS
diff -Naur xorg-server-1.20.11/hw/xwayland/xwayland-output.c xorg-server-1.20.11/hw/xwayland/xwayland-output.c
--- xorg-server-1.20.11/hw/xwayland/xwayland-output.c	2021-04-13 17:11:40.000000000 +0300
+++ xorg-server-1.20.11/hw/xwayland/xwayland-output.c	2021-08-14 10:49:27.882039515 +0300
@@ -187,6 +187,9 @@
     if (!xwl_screen->rootless && xwl_screen->screen->root)
         update_backing_pixmaps (xwl_screen, width, height);
 
+    if (!xwl_screen->rootless && xwl_screen->screen->root)
+        update_backing_pixmaps (xwl_screen, width, height);
+
     xwl_screen->width = width;
     xwl_screen->height = height;
     xwl_screen->screen->width = width;
diff -Naur xorg-server-1.20.11/hw/xwayland/xwayland-shm.c xorg-server-1.20.11/hw/xwayland/xwayland-shm.c
--- xorg-server-1.20.11/hw/xwayland/xwayland-shm.c	2021-04-13 17:11:40.000000000 +0300
+++ xorg-server-1.20.11/hw/xwayland/xwayland-shm.c	2021-08-14 10:48:34.155323981 +0300
@@ -40,12 +40,17 @@
 #include <string.h>
 #include <stdlib.h>
 
+#ifdef __ANDROID__
+#include "ashmem.h"
+#endif
+
 struct xwl_pixmap {
     struct wl_buffer *buffer;
     void *data;
     size_t size;
 };
 
+#ifndef __ANDROID__
 #ifndef HAVE_MKOSTEMP
 static int
 set_cloexec_or_close(int fd)
@@ -172,6 +177,37 @@
     return fd;
 }
 
+#else // __ANDROID__
+
+static int
+os_create_anonymous_file(off_t size) {
+	int fd, ret;
+    long flags;
+
+	fd = open("/dev/ashmem", O_RDWR);
+	if (fd < 0)
+		return fd;
+
+	ret = ioctl(fd, ASHMEM_SET_SIZE, size);
+	if (ret < 0)
+		goto err;
+
+    flags = fcntl(fd, F_GETFD);
+    if (flags == -1)
+        goto err;
+
+    if (fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1)
+        goto err;
+
+	return fd;
+
+err:
+	close(fd);
+	return ret;
+}
+
+#endif // __ANDROID__
+
 static uint32_t
 shm_format_for_depth(int depth)
 {
@@ -202,6 +238,7 @@
     int fd;
 
     if (hint == CREATE_PIXMAP_USAGE_GLYPH_PICTURE ||
+        (!xwl_screen->rootless && hint != CREATE_PIXMAP_USAGE_BACKING_PIXMAP) ||
         (width == 0 && height == 0) || depth < 15)
         return fbCreatePixmap(screen, width, height, depth, hint);
 
diff -Naur xorg-server-1.20.11/miext/sync/meson.build xorg-server-1.20.11/miext/sync/meson.build
--- xorg-server-1.20.11/miext/sync/meson.build	2021-04-13 17:11:40.000000000 +0300
+++ xorg-server-1.20.11/miext/sync/meson.build	2021-08-14 10:48:34.155323981 +0300
@@ -1,6 +1,7 @@
 srcs_miext_sync = [
     'misync.c',
     'misyncfd.c',
+    'misyncshm.c',
 ]
 
 hdrs_miext_sync = [
@@ -10,10 +11,6 @@
     'misyncstr.h',
 ]
 
-if build_dri3
-    srcs_miext_sync += 'misyncshm.c'
-endif
-
 libxserver_miext_sync = static_library('libxserver_miext_sync',
     srcs_miext_sync,
     include_directories: inc,
