--- ./jdk/src/solaris/native/sun/nio/fs/UnixNativeDispatcher.c.orig	2021-05-11 06:47:42.139946825 +0000
+++ ./jdk/src/solaris/native/sun/nio/fs/UnixNativeDispatcher.c	2021-05-11 06:48:06.407640393 +0000
@@ -1020,41 +1020,6 @@
     return result;
 }
 
-#ifdef __ANDROID__
-/*
- * TODO: Android lacks support for the methods listed below.  In it's place are
- * alternatives that use existing Android functionality, but lack reentrant
- * support.  Determine if the following are the most suitable alternatives.
- *
- */
-int getgrgid_r(gid_t gid, struct group* grp, char* buf, size_t buflen, struct group** result) {
-
-  *result = NULL;
-  errno = 0;
-  grp = getgrgid(gid);
-  if (grp == NULL) {
-        return errno;
-  }
-  // buf not used by caller (see below)
-  *result = grp;
-  return 0;
-}
-
-int getgrnam_r(const char *name, struct group* grp, char* buf, size_t buflen, struct group** result) {
-
-  *result = NULL;
-  errno = 0;
-  grp = getgrnam(name);
-  if (grp == NULL) {
-        return errno;
-  }
-  // buf not used by caller (see below)
-  *result = grp;
-  return 0;
-
-}
-#endif // __ANDROID__
-
 JNIEXPORT jbyteArray JNICALL
 Java_sun_nio_fs_UnixNativeDispatcher_getgrgid(JNIEnv* env, jclass this, jint gid)
 {
