--- ./make/common/NativeCompilation.gmk.orig	2021-05-11 06:20:03.404504209 +0000
+++ ./make/common/NativeCompilation.gmk	2021-05-11 06:20:16.972559543 +0000
@@ -107,7 +107,7 @@
 	    $$($1_$2_COMP) $$($1_$2_FLAGS) $$($1_$2_DEP_FLAG) $$($1_$2_DEP).tmp $(CC_OUT_OPTION)$$($1_$2_OBJ) $2
 	    $(SED) 's|^$$(@F):|$$@:|' $$($1_$2_DEP).tmp > $$($1_$2_DEP)
           else
-	    $$($1_$2_COMP) $$($1_$2_FLAGS) $$($1_$2_DEP_FLAG) $$($1_$2_DEP) $(CC_OUT_OPTION)$$($1_$2_OBJ) $2
+	    $$($1_$2_COMP) $$($1_$2_FLAGS) -MD $(CC_OUT_OPTION)$$($1_$2_OBJ) $2
           endif
         endif
         # The Visual Studio compiler lacks a feature for generating make dependencies, but by
